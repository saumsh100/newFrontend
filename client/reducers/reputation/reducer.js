import { fromJS } from 'immutable';

import {
  FETCH_REPUTATION,
  FETCH_REPUTATION_SUCCESS
} from './constants';

const initialState = fromJS({
  status: 'loading', // set only to 'loading', 'success' or 'error'
  lastFetched: null,
  data: {
    sourcesNotFound: null,
    sourcesFound: null,
    listingScore: null,
    listingPointScore: {
      industryAverage: null,
      industryLeadersAverage: null,
      pointScore: null
    },
    citationsFound: null,
    sourcesFoundWithErrors: null,
    accuracyScore: null
  }
})

function reputationReducer(state = initialState, action) {
  switch (action.type) {
    case FETCH_REPUTATION:
      return state.set('status', 'loading')
    case FETCH_REPUTATION_SUCCESS:
      return state
        .set('status', 'success')
        .set('lastFetched', (new Date()).toLocaleString())
        .set('data', action.data)
    default:
      return state
  }
}

export default reputationReducer;