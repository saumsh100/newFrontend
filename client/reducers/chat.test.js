
import { fromJS } from 'immutable';
import reducer, {
  initialState,
  setUnreadChats,
  setChatMessages,
  setChatPoC,
  setLockedChats,
  setNewChat,
  setSelectedChat,
  setTotalChatMessages,
  updateChatId,
} from './chat';

const setUnreadChatsAction = setUnreadChats(['123', '456', '789']);
const setChatMessagesAction = setChatMessages([
  {
    id: '123',
    body: 'test',
  },
  {
    id: '321',
    body: 'test 1',
  },
]);
const setChatPoCActionNoPayload = setChatPoC();
const setChatPoCActionId = setChatPoC({ id: '333' });
const setLockedChatsAction = setLockedChats(['123']);
const setNewChatAction = setNewChat({ patientId: '333' });
const setSelectedChatAction = setSelectedChat(fromJS({
  id: '123',
  patientId: '333',
}));
const setTotalChatMessagesAction = setTotalChatMessages(13);
const updateChatIdAction = updateChatId();

describe('chat reducer', () => {
  test('action creator works', () => {
    expect(typeof setUnreadChats).toBe('function');
    expect(typeof setChatMessages).toBe('function');
    expect(typeof setChatPoC).toBe('function');
    expect(typeof setLockedChats).toBe('function');
    expect(typeof setNewChat).toBe('function');
    expect(typeof setSelectedChat).toBe('function');
    expect(typeof setTotalChatMessages).toBe('function');
    expect(typeof updateChatId).toBe('function');
  });

  test('setUnreadChats', () => {
    expect(setUnreadChatsAction).toMatchSnapshot();
  });

  test('setChatMessages', () => {
    expect(setChatMessagesAction).toMatchSnapshot();
  });

  test('setChatPoC no payload', () => {
    expect(setChatPoCActionNoPayload).toMatchSnapshot();
  });

  test('setChatPoC', () => {
    expect(setChatPoCActionId).toMatchSnapshot();
  });

  test('setLockedChats', () => {
    expect(setLockedChatsAction).toMatchSnapshot();
  });

  test('setNewChat', () => {
    expect(setNewChatAction).toMatchSnapshot();
  });

  test('setSelectedChat', () => {
    expect(setSelectedChatAction).toMatchSnapshot();
  });

  test('setTotalChatMessages', () => {
    expect(setTotalChatMessagesAction).toMatchSnapshot();
  });

  test('updateChatId', () => {
    expect(updateChatIdAction).toMatchSnapshot();
  });

  describe('reducers', () => {
    test('base', () => {
      expect(reducer(initialState, {})).toEqual(initialState);
    });

    test('setUnreadChats', () => {
      expect(reducer(initialState, setUnreadChatsAction)).toMatchSnapshot();
    });

    test('setChatMessages', () => {
      expect(reducer(initialState, setChatMessagesAction)).toMatchSnapshot();
    });

    test('setChatPoC no payload', () => {
      expect(reducer(initialState, setChatPoCActionNoPayload)).toMatchSnapshot();
    });

    test('setChatPoC with selected chat', () => {
      const stateWithSelectedChat = reducer(initialState, setSelectedChatAction);
      const PoCset = reducer(stateWithSelectedChat, setChatPoCActionId);
      expect(PoCset).toMatchSnapshot();
    });

    test('setChatPoC with newChat', () => {
      const newChatPocAction = setNewChat({ patientId: '333' });
      const stateWithNewChat = reducer(initialState, newChatPocAction);
      const PoCset = reducer(stateWithNewChat, setChatPoCActionId);
      expect(PoCset).toMatchSnapshot();
    });

    test('setNewChat', () => {
      expect(reducer(initialState, setNewChatAction)).toMatchSnapshot();
    });

    test('setLockedChats', () => {
      expect(reducer(initialState, setLockedChatsAction)).toMatchSnapshot();
    });

    test('setSelectedChat', () => {
      expect(reducer(initialState, setSelectedChatAction)).toMatchSnapshot();
    });

    test('setTotalChatMessages', () => {
      expect(reducer(initialState, setTotalChatMessagesAction)).toMatchSnapshot();
    });

    test('updateChatId', () => {
      expect(reducer(initialState, updateChatIdAction)).toMatchSnapshot();
    });

    test('updateChatId with selected chat', () => {
      const stateWithSelectedChat = reducer(initialState, setSelectedChatAction);
      const updatedChat = reducer(stateWithSelectedChat, updateChatIdAction);
      expect(updatedChat).toMatchSnapshot();
    });
  });
});
