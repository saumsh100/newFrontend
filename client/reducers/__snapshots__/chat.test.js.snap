// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat reducer reducers setChatMessages 1`] = `
Immutable.Map {
  "chatMessages": Array [
    Object {
      "body": "test",
      "id": "123",
    },
    Object {
      "body": "test 1",
      "id": "321",
    },
  ],
  "selectedPatientId": null,
  "totalChatMessages": 0,
  "unreadChats": Array [],
  "chatPoC": null,
  "selectedChatId": null,
  "lockedChats": Array [],
  "selectedChat": null,
  "chatList": Object {},
  "newChat": null,
  "isPoC": null,
}
`;

exports[`chat reducer reducers setChatPoC no payload 1`] = `
Immutable.Map {
  "chatMessages": Array [],
  "selectedPatientId": null,
  "totalChatMessages": 0,
  "unreadChats": Array [],
  "chatPoC": Object {},
  "selectedChatId": null,
  "lockedChats": Array [],
  "selectedChat": null,
  "chatList": Object {},
  "newChat": null,
  "isPoC": true,
}
`;

exports[`chat reducer reducers setChatPoC with newChat 1`] = `
Immutable.Map {
  "chatMessages": Array [],
  "selectedPatientId": null,
  "totalChatMessages": 0,
  "unreadChats": Array [],
  "chatPoC": Object {
    "id": "333",
  },
  "selectedChatId": null,
  "lockedChats": Array [],
  "selectedChat": null,
  "chatList": Object {},
  "newChat": Object {
    "patientId": "333",
  },
  "isPoC": true,
}
`;

exports[`chat reducer reducers setChatPoC with selected chat 1`] = `
Immutable.Map {
  "chatMessages": Array [],
  "selectedPatientId": null,
  "totalChatMessages": 0,
  "unreadChats": Array [],
  "chatPoC": Object {
    "id": "333",
  },
  "selectedChatId": null,
  "lockedChats": Array [],
  "selectedChat": Immutable.Map {
    "id": "123",
    "patientId": "333",
  },
  "chatList": Object {},
  "newChat": null,
  "isPoC": true,
}
`;

exports[`chat reducer reducers setLockedChats 1`] = `
Immutable.Map {
  "chatMessages": Array [],
  "selectedPatientId": null,
  "totalChatMessages": 0,
  "unreadChats": Array [],
  "chatPoC": null,
  "selectedChatId": null,
  "lockedChats": Array [
    "123",
  ],
  "selectedChat": null,
  "chatList": Object {},
  "newChat": null,
  "isPoC": null,
}
`;

exports[`chat reducer reducers setNewChat 1`] = `
Immutable.Map {
  "chatMessages": Array [],
  "selectedPatientId": null,
  "totalChatMessages": 0,
  "unreadChats": Array [],
  "chatPoC": null,
  "selectedChatId": null,
  "lockedChats": Array [],
  "selectedChat": null,
  "chatList": Object {},
  "newChat": Object {
    "patientId": "333",
  },
  "isPoC": null,
}
`;

exports[`chat reducer reducers setSelectedChat 1`] = `
Immutable.Map {
  "chatMessages": Array [],
  "selectedPatientId": null,
  "totalChatMessages": 0,
  "unreadChats": Array [],
  "chatPoC": null,
  "selectedChatId": null,
  "lockedChats": Array [],
  "selectedChat": Immutable.Map {
    "id": "123",
    "patientId": "333",
  },
  "chatList": Object {},
  "newChat": null,
  "isPoC": null,
}
`;

exports[`chat reducer reducers setTotalChatMessages 1`] = `
Immutable.Map {
  "chatMessages": Array [],
  "selectedPatientId": null,
  "totalChatMessages": 13,
  "unreadChats": Array [],
  "chatPoC": null,
  "selectedChatId": null,
  "lockedChats": Array [],
  "selectedChat": null,
  "chatList": Object {},
  "newChat": null,
  "isPoC": null,
}
`;

exports[`chat reducer reducers setUnreadChats 1`] = `
Immutable.Map {
  "chatMessages": Array [],
  "selectedPatientId": null,
  "totalChatMessages": 0,
  "unreadChats": Array [
    "123",
    "456",
    "789",
  ],
  "chatPoC": null,
  "selectedChatId": null,
  "lockedChats": Array [],
  "selectedChat": null,
  "chatList": Object {},
  "newChat": null,
  "isPoC": null,
}
`;

exports[`chat reducer reducers updateChatId 1`] = `
Immutable.Map {
  "chatMessages": Array [],
  "selectedPatientId": null,
  "totalChatMessages": 0,
  "unreadChats": Array [],
  "chatPoC": null,
  "selectedChatId": null,
  "lockedChats": Array [],
  "selectedChat": null,
  "chatList": Object {},
  "newChat": null,
  "isPoC": null,
}
`;

exports[`chat reducer reducers updateChatId with selected chat 1`] = `
Immutable.Map {
  "chatMessages": Array [],
  "selectedPatientId": null,
  "totalChatMessages": 0,
  "unreadChats": Array [],
  "chatPoC": null,
  "selectedChatId": "123",
  "lockedChats": Array [],
  "selectedChat": Immutable.Map {
    "id": "123",
    "patientId": "333",
  },
  "chatList": Object {},
  "newChat": null,
  "isPoC": null,
}
`;

exports[`chat reducer setChatMessages 1`] = `
Object {
  "payload": Array [
    Object {
      "body": "test",
      "id": "123",
    },
    Object {
      "body": "test 1",
      "id": "321",
    },
  ],
  "type": "@chat/SET_CHAT_MESSAGES",
}
`;

exports[`chat reducer setChatPoC 1`] = `
Object {
  "payload": Object {
    "id": "333",
  },
  "type": "@chat/SET_CHAT_POC",
}
`;

exports[`chat reducer setChatPoC no payload 1`] = `
Object {
  "payload": undefined,
  "type": "@chat/SET_CHAT_POC",
}
`;

exports[`chat reducer setLockedChats 1`] = `
Object {
  "payload": Array [
    "123",
  ],
  "type": "@chat/SET_LOCKED_CHATS",
}
`;

exports[`chat reducer setNewChat 1`] = `
Object {
  "payload": Object {
    "patientId": "333",
  },
  "type": "@chat/SET_NEW_CHAT",
}
`;

exports[`chat reducer setSelectedChat 1`] = `
Object {
  "payload": Immutable.Map {
    "id": "123",
    "patientId": "333",
  },
  "type": "@chat/SET_SELECTED_CHAT",
}
`;

exports[`chat reducer setTotalChatMessages 1`] = `
Object {
  "payload": 13,
  "type": "@chat/SET_TOTAL_CHAT_MESSAGES",
}
`;

exports[`chat reducer setUnreadChats 1`] = `
Object {
  "payload": Array [
    "123",
    "456",
    "789",
  ],
  "type": "@chat/SET_UNREAD_CHATS",
}
`;

exports[`chat reducer updateChatId 1`] = `
Object {
  "payload": undefined,
  "type": "@chat/UPDATE_CHAT_ID",
}
`;
